featurebyte/schema/worker/task/feature_list_create.py

```diff
@@ -21,10 +22,35 @@ class FeatureParameters(FeatureByteBaseModel):
     name: NameStr
 
 
+def feature_params_discriminator(value: Any) -> Literal["feature_ids", "feature_params"]:
+    """
+    Discriminator for feature parameters
+
+    Parameters
+    ----------
+    value: Any
+        Input value
+
+    Returns
+    -------
+    Literal["feature_ids", "feature_params"]
+    """
+    if isinstance(value, list):
+        if value:
+            return feature_params_discriminator(value[0])
+        return "feature_ids"
+    if isinstance(value, (dict, FeatureParameters)):
+        return "feature_params"
+    return "feature_ids"
+
+
 class FeaturesParameters(FeatureByteBaseModel):
     """Feature list feature parameters"""
 
-    features: Union[List[FeatureParameters], List[PydanticObjectId]]
+    features: Union[
+        Annotated[List[FeatureParameters], Tag("feature_params")],
+        Annotated[List[PydanticObjectId], Tag("feature_ids")],
+    ] = Field(discriminator=Discriminator(feature_params_discriminator), min_length=1)
```

Comment:
`features: Union[List[FeatureParameters], List[PydanticObjectId]]` is no longer work in pydantic V2. In pydantic V2, a discrimination function is required to deserialize.